awx/main/conf.py

```diff
@@ -795,6 +795,94 @@
     category_slug='bulk',
 )
 
+register(
+    'SYSTEM_TASK_ABS_CPU',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    label=_('Override CPU for capacity calculations for all control instances. Accepts formats equivalent to kubernetes resource requests like 4 or 4000m'),
+    help_text=_('Override CPU for capacity calculations for all control instances. Accepts formats equivalent to kubernetes resource requests like 4 or 4000m'),
+    category=_('Task'),
+    category_slug='task',
+)
+
+register(
+    'SYSTEM_TASK_ABS_MEM',
+    field_class=fields.CharField,
+    default='',
+    allow_blank=True,
+    label=_('Override Memory for capacity calculations for **ALL** instances. Accepts formats equivalent to kubernetes resource requests like 1Gi or 500Mi'),
+    help_text=_(
+        'Override Memory for capacity calculations for **ALL** instances. Accepts formats equivalent to kubernetes resource requests like 1Gi or 500Mi'
```

Comment:
The first sentence is hard to read.

It would also help _immensely_ to spell out the behavior, if the setting is _not_ set. That behavior would be that the heartbeat task will dynamically check to find out how much memory and CPU is available.

But it's more complicated for OCP, because the expectation is that the awx-operator (IIRC) will define a value in settings, which will take precedence over the `default=''` here (again, I _think_, needs confirmation). So in that case, not setting it doesn't necessarily result in a garbage value, but an appropriate value, and users should really think twice before changing it here.

Reply:
in ocp if its not set -- then it will read from node. e.g. the way the operator communicates anything that is not garbage, is by setting this setting. So you would see it. But maybe I can spell out that in ocp, this setting is controlled by operator if resource limits are set on the task container